### DISCUSSION TITLE
Can't seem to get double-wide powerline symbols to work

### DISCUSSION THREAD

**Discussion by Machine-Maker** [kovidgoyal/kitty#6299] [Q&A]
I have the fonts setup correctly (as far as I know). I'm not using a patched font, just using JetBrains Mono as the normal font and then I have
```
symbol_map U+23FB-U+23FE,U+2665,U+26A1,U+2B58,U+E000-U+E00A,U+E0A0-U+E0A3,U+E0B0-U+E0D4,U+E200-U+E2A9,U+E300-U+E3E3,U+E5FA-U+E6AA,U+E700-U+E7C5,U+EA60-U+EBEB,U+F000-U+F2E0,U+F300-U+F32F,U+F400-U+F4A9,U+F500-U+F8FF,U+F0001-U+F1AF0 Symbols Nerd Font Mono
```
to set the font for the symbols.

However, the powerline double wide glyphs, [shown here](https://github.com/ryanoasis/powerline-extra-symbols#glyphs), do not show up as double wide, even if there is a whitespace after them. I was reading the kitty docs, and saw the `narrow_symbols` setting, so I tried 
```
narrow_symbols U+E0B8,U+30B9 2
```
to set just 2 of those glyphs to 2 width, but that didnt work.
![image](https://github.com/kovidgoyal/kitty/assets/15055071/2b999a42-7431-48cb-b050-a1a51a7d5737)

I've found similar questions people have asked, but they've all been pointed to using the narrow_symbols setting, which either isn't the solution, or I'm using it incorrectly.

EDIT: It doesn't seem to be all nerd font icons that are affected, like I noticed this
<img width="117" alt="image" src="https://github.com/kovidgoyal/kitty/assets/15055071/87351c89-4482-44c3-9ade-5a57102ec5f8">
When I remove the space after the icon, then it does shrink to a single width, which reflects what I read on the documentation. 

2nd EDIT: It seems to literally just be these 8 icons that are affected, all the rest follow the "rules". 
<img width="771" alt="image" src="https://github.com/kovidgoyal/kitty/assets/15055071/5aa1a649-fb47-4c5f-9710-eb0e302db363">

**Comment by kovidgoyal** [comment]
Use --debug-font-fallback to make sure the symbol is coming from the
font you want it to. If it isnt use symbol_map in kitty.conf to fix it.

**Reply by Machine-Maker** [reply]
```
[143 23:24:48.743796] Preloaded font faces:
[143 23:24:48.743821] normal face: /Users/jake/Library/Fonts/JetBrainsMono-Regular.ttf
[143 23:24:48.743826] bold face: /Users/jake/Library/Fonts/JetBrainsMono-SemiBold.ttf
[143 23:24:48.743831] italic face: /Users/jake/Library/Fonts/JetBrainsMono-Italic.ttf
[143 23:24:48.743835] bi face: /Users/jake/Library/Fonts/JetBrainsMono-SemiBoldItalic.ttf
[143 23:24:48.743840] Symbol map faces:
[143 23:24:48.743844] /Users/jake/Library/Fonts/Symbols-2048-em Nerd Font Complete Mono.ttf
```
It's coming from the symbol font, and I am manually setting it using `symbol_map`. 
```
symbol_map U+23FB-U+23FE,U+2665,U+26A1,U+2B58,U+E000-U+E00A,U+E0A0-U+E0A3,U+E0B0-U+E0D4,U+E200-U+E2A9,U+E300-U+E3E3,U+E5FA-U+E6AA,U+E700-U+E7C5,U+EA60-U+EBEB,U+F000-U+F2E0,U+F300-U+F32F,U+F400-U+F4A9,U+F500-U+F8FF,U+F0001-U+F1AF0 Symbols Nerd Font Mono

```

**Comment by kovidgoyal** [comment]
Those are separators, they dont come from fonts, they are drawn programatically by kitty to always line up correctly. And they will never be wide, indeed they are not supposed to be wide. I dont know where you get the idea that they are supposed to be wide. See box_drawing.py for how they are drawn.

**Reply by Machine-Maker** [reply]
Oh, I guess I misunderstood what they were. I'm migrating from using iTerm on mac, and the characters are 2-wide there. 
<img width="217" alt="image" src="https://github.com/kovidgoyal/kitty/assets/15055071/e362160b-5cb0-4b24-b6d8-cd72b0d1e8fd"> (from iTerm)

They also look double width inside macos' font book (as shown in an earlier image).

They also are double wide, like other icons, on the nerd font cheatsheet. 
<img width="172" alt="image" src="https://github.com/kovidgoyal/kitty/assets/15055071/75880d5d-07a0-4210-8938-01ae1abb6911">

And I feel like most of the images I see of them being used, are as double width, mainly for the full right-isosceles triangle.

**Reply by kovidgoyal** [reply]
If you can find a reference actually stating they are supposed to be wide I will reconsider, but they have always been rendered single spaced in every context I have seen them. And kitty has drawn them thus for many many years. You are the first person that has expected them to be wide. I am fairly sure that if I changed them there would be a fair few surprised people.

**Reply by Machine-Maker** [reply]
This isn't really a "reference", but this was the first time I realized there could be a difference. https://github.com/akinsho/bufferline.nvim/blob/main/doc/bufferline.txt#L193-L208.

Basically, the "padded" option add an extra space so if the char is 2-wide, it looks right. But they have options for both, so I guess some people expect one, some the other. 

It's fine, I just thought it was an issue, I'll just have to re-configure some things.
